src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -296,648 +454,673 @@ def build_process_costs(
                 f"1990 to 2020."
             )
 
-        if total_purchase_cost is None:
-            self.get_total_BEC(CE_index_year, watertap_blocks)
-        else:
-            self.total_BEC = Var(
-                initialize=total_purchase_cost,
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
-            self.total_BEC.fix()
+        if (
+            fixed_OM is False and calculate_NPV is True
+        ):  # NPV method invoked with fixed inputs
+            self.calculate_NPV(fixed_OM, variable_OM)
 
-        # define variables
-        if Lang_factor is None:
-            # initialize parameters from specified percentages
-            self.piping_materials_and_labor_percentage = Param(
-                mutable=True,
-                initialize=piping_materials_and_labor_percentage / 100,
-                doc="Piping, materials and labor",
-            )
+        else:  # continue on with building the plant costs
 
-            self.electrical_materials_and_labor_percentage = Param(
-                mutable=True,
-                initialize=electrical_materials_and_labor_percentage / 100,
-                doc="Electrical, materials and labor",
-            )
+            if total_purchase_cost is None:
+                self.get_total_BEC(CE_index_year, watertap_blocks)
+            else:
+                self.total_BEC = Var(
+                    initialize=total_purchase_cost,
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
+                self.total_BEC.fix()
 
-            self.instrumentation_percentage = Param(
-                mutable=True,
-                initialize=instrumentation_percentage / 100,
-                doc="Instrumentation",
-            )
+            # define variables
+            if Lang_factor is None:
+                # initialize parameters from specified percentages
+                self.piping_materials_and_labor_percentage = Param(
+                    mutable=True,
+                    initialize=piping_materials_and_labor_percentage / 100,
+                    doc="Piping, materials and labor",
```

Comment:
These doc strings do not really describe what this value is. Also, I note you are converting from a percentage value as an input to a decimal value for the parameter. I think this is a bad idea, as it creates a separation between the user input and the actual model which will result in confusion (i.e. a user might go to update one of these manually and expect to provide the value in the same basis they used when declaring the class).